lib/lightning_web/live/workflow_live/ai_assistant_component.ex

```diff
@@ -71,10 +76,18 @@ defmodule LightningWeb.WorkflowLive.AiAssistantComponent do
 
   def handle_event("send_message", %{"content" => content}, socket) do
     if socket.assigns.can_edit_workflow do
-      {:noreply,
-       socket
-       |> assign(error_message: nil)
-       |> save_message(socket.assigns.action, content)}
+      %{project_id: project_id, action: action} = socket.assigns
+
+      case Limiter.validate_quota(project_id) do
+        :ok ->
+          save_message(socket, action, content)
+
+        {:error, :too_many_queries, message} ->
+          notify_limit_error(socket, message)
+      end
+      |> then(fn socket ->
```

Comment:
This will clear the `error_message` assign which has been added by `save_message`. It should be at the top, before `save_message` is called